examples/stock/company.v.py

```diff
@@ -25,29 +27,23 @@ def buy_stock():
     buy_order = msg.value / self.price # rounds down
 
     # There are enough shares to buy
-    assert self.avail_shares >= buy_order
+    assert self.holdings[self.company] >= buy_order
     
     # Take the shares off the market and give to stockholder
-    self.avail_shares -= buy_order
-    if self.holdings[msg.sender]:
-        # Account exists, so add to that account
-        self.holdings[msg.sender] += buy_order
-    else:
-        # Not currently a stockholder, so add them
-        self.holdings[msg.sender] = buy_order
+    self.holdings[self.company] -= buy_order
+    self.holdings[msg.sender] += buy_order
 
```

Comment:
Maybe it's too complex for this example but should partial buys be an option? e.g. I wanna buy 10 shares and there only 5 shares left.  As it is right now it'll just throw, it might be nice if I could try to buy 10 and get 5 with a half refund.